packages/framework/src/Component/Elasticsearch/ElasticsearchStructureManager.php

```diff
@@ -79,57 +109,141 @@ public function getAliasName(int $domainId, string $index): string
      */
     public function createIndex(int $domainId, string $index)
     {
-        $definition = $this->getDefinition($domainId, $index);
+        $definition = $this->getStructureDefinition($domainId, $index);
         $indexes = $this->client->indices();
-        $aliasName = $this->getAliasName($domainId, $index);
-        $indexName = $this->getIndexName($domainId, $index);
+        $indexName = $this->getVersionedIndexName($domainId, $index);
         if ($indexes->exists(['index' => $indexName])) {
+            if ($this->buildVersion === 0) {
+                throw new ElasticsearchStructureException('Index %s already exists. Please start using build version or delete current index before creating a new one.');
```

Comment:
The placeholder for `$indexName` doesn't get replaced here.

Maybe the code would get clearer if you rewrote it with only one `sprintf` and one `throw`:
```php
$message = sprintf('Index %s already exists', $indexName);
if ($this->buildVersion === 0) {
    $message .= ' Please start using build version or delete current index before creating a new one.';
}
throw new ElasticsearchStructureException($message);
```